sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.cassandra;
+
+import static org.junit.Assert.assertEquals;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Metadata;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.easymock.EasyMock;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests on {@link CassandraServiceImplTest}.
+ */
+public class CassandraServiceImplTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CassandraServiceImplTest.class);
+
+  private static final String MURMUR3_PARTITIONER = "org.apache.cassandra.dht.Murmur3Partitioner";
+  private static final String RANDOM_PARTITIONER = "org.apache.cassandra.dht.RandomPartitioner";
+
+  private Cluster createClusterMock(String partitioner) {
+    Metadata metadata = EasyMock.createMock(Metadata.class);
+    EasyMock.expect(metadata.getPartitioner()).andReturn(partitioner).anyTimes();
+    Cluster cluster = EasyMock.createMock(Cluster.class);
+    EasyMock.expect(cluster.getMetadata()).andReturn(metadata).anyTimes();
+    EasyMock.replay(metadata);
+    EasyMock.replay(cluster);
+    return cluster;
+  }
+
+  @Test
+  public void testDistanceWithRandomPartitioner() throws Exception {
+    Cluster cluster = createClusterMock(RANDOM_PARTITIONER);
+    BigInteger distance = CassandraServiceImpl.distance(cluster, 10L, 100L);
+
+    assertEquals(new BigInteger("90"), distance);
+
+    distance = CassandraServiceImpl.distance(cluster, 100L, 10L);
+    assertEquals(new BigInteger("170141183460469231731687303715884105639"), distance);
+  }
+
+  @Test
+  public void testRingFractionWithRandomPartitioner() throws Exception {
+    Cluster cluster = createClusterMock(RANDOM_PARTITIONER);
+    List<CassandraServiceImpl.TokenRange> tokenRanges = new ArrayList<>();
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(0, 100, 0, 99));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(1, 100, 100, 199));
+    BigDecimal ringFraction = CassandraServiceImpl.getRingFraction(cluster, tokenRanges);
+
+    System.out.println(ringFraction);
```

Comment:
I don't think this is a sufficient level of testing. We need to test that the value is correct too.